File: HSNcode/src/main/java/HSNcode/api/HSNcode/model/ShipperDetails.java
package HSNcode.api.HSNcode.model;

import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "shipper_details")
public class ShipperDetails {

    // Getters and Setters
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "shipper_id")
    private Long shipperId;

    @Column(name = "shipperfname")
    private String shipperfname;

    @Column(name = "shipperlname")
    private String shipperlname;

    @Column(name = "shipper_address1")
    private String shipperAddress1;

    @Column(name = "shipper_address2")
    private String shipperAddress2;

    @Column(name = "shipper_city")
    private String shipperCity;

    @Column(name = "shipper_state")
    private String shipperState;

    @Column(name = "shipper_country")
    private String shipperCountry;

    @Column(name = "shipper_email")
    private String shipperEmail;

    @Column(name = "shipper_phone")
    private String shipperPhone;

    @Column(name = "shippercin")
    private String shipperCIN;

    @Column(name = "shipperuoeno")
    private String shipperUOENo;

    public void setShipperId(Long shipperId) {
        this.shipperId = shipperId;
    }

    public void setShipperfname(String shipperfname) {
        this.shipperfname = shipperfname;
    }

    public void setShipperlname(String shipperlname) {
        this.shipperlname = shipperlname;
    }

    public void setShipperAddress1(String shipperAddress1) {
        this.shipperAddress1 = shipperAddress1;
    }

    public void setShipperAddress2(String shipperAddress2) {
        this.shipperAddress2 = shipperAddress2;
    }

    public void setShipperCity(String shipperCity) {
        this.shipperCity = shipperCity;
    }

    public void setShipperState(String shipperState) {
        this.shipperState = shipperState;
    }

    public void setShipperCountry(String shipperCountry) {
        this.shipperCountry = shipperCountry;
    }

    public void setShipperEmail(String shipperEmail) {
        this.shipperEmail = shipperEmail;
    }

    public void setShipperPhone(String shipperPhone) {
        this.shipperPhone = shipperPhone;
    }

    public void setShipperCIN(String shipperCIN) {
        this.shipperCIN = shipperCIN;
    }

    public void setShipperUOENo(String shipperUOENo) {
        this.shipperUOENo = shipperUOENo;
    }
}

File: fronend/src/components/ShipperDetails.js
import React, { useState } from "react";
import NameAndAddressDetails from "./NameAndAddressDetails";
import axios from "axios";

const ShipperDetails = ({ nextStep }) => {
  const [shipperfname, setShipperFName] = useState("");
  const [shipperlname, setShipperLName] = useState("");
  const [shipper_address1, setShipperAddress1] = useState("");
  const [shipper_address2, setShipperAddress2] = useState("");
  const [shipper_city, setShipperCity] = useState("");
  const [shipper_state, setShipperState] = useState("");
  const [shipper_country, setShipperCountry] = useState("");
  const [shipper_email, setShipperEmail] = useState("");
  const [shipper_phone, setShipperPhone] = useState("");
  const [shippercin, setShipperCIN] = useState("");
  const [shipperuoeno, setShipperUOENo] = useState("");
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);
  const [success, setSuccess] = useState(false);

  const handleSaveShipperDetails = async () => {
    const shipperData = {
      shipperfname,
      shipperlname,
      shipperAddress1: shipper_address1, // Map React field to Java entity field
    shipperAddress2: shipper_address2, // Map React field to Java entity field
    shipperCity: shipper_city, // Map React field to Java entity field
    shipperState: shipper_state, // Map React field to Java entity field
    shipperCountry: shipper_country, // Map React field to Java entity field
    shipperEmail: shipper_email, // Map React field to Java entity field
    shipperPhone: shipper_phone, // Map React field to Java entity field
    shipperCIN: shippercin, // Map React field to Java entity field
    shipperUOENo: shipperuoeno, // Map React field to Java entity field
    };

    console.log("Shipper Data:", shipperData); // Debugging log

    try {
      setLoading(true);
      setError(null);
      const response = await axios.post("/api/bill-of-lading/shipper", shipperData);
      setSuccess(true);
      console.log("Shipper details saved successfully:", response.data);
      nextStep(1);
    } catch (err) {
      setError(err.response?.data?.message || "Failed to save shipper details.");
      console.error("Error saving shipper details:", err);
    } finally {
      setLoading(false);
    }
  };

  return (
    <div className="step-container">
      <div className="step">
        <h2>Shipper Details</h2>
        <NameAndAddressDetails
          prefix="shipper"
          fName={shipperfname}
          lName={shipperlname}
          address1={shipper_address1}
          address2={shipper_address2}
          city={shipper_city}
          state={shipper_state}
          country={shipper_country}
          email={shipper_email}
          phone={shipper_phone}
          onFNameChange={setShipperFName}
          onLNameChange={setShipperLName}
          onAddress1Change={setShipperAddress1}
          onAddress2Change={setShipperAddress2}
          onCityChange={setShipperCity}
          onStateChange={setShipperState}
          onCountryChange={setShipperCountry}
          onEmailChange={setShipperEmail}
          onPhoneChange={setShipperPhone}
        />
        <label htmlFor="shippercin">CIN:</label>
        <input
          type="text"
          id="shippercin"
          value={shippercin}
          onChange={(e) => setShipperCIN(e.target.value)}
          required
        />
        <label htmlFor="shipperuoeno">UOE Number:</label>
        <input
          type="text"
          id="shipperuoeno"
          value={shipperuoeno}
          onChange={(e) => setShipperUOENo(e.target.value)}
          required
        />
        <div className="navigation">
          <button
            type="button"
            className="next"
            onClick={handleSaveShipperDetails}
            disabled={loading}
          >
            {loading ? "Saving..." : "Next"}
          </button>
        </div>
        {error && <p className="error">{error}</p>}
        {success && <p className="success">Shipper details saved successfully!</p>}
      </div>
    </div>
  );
};

export default ShipperDetails;
